ARCs/arc-0014.md

```diff
@@ -0,0 +1,247 @@
+---
+arc: 14
+title: Authentication with Algorand accounts
+description: Use Algorand accounts to authenticate with third-party services
+author: Stefano De Angelis (@deanstef)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/42
+status: Draft
+type: Meta
+created: 2022-04-05
+---
+
+# Authentication with Algorand accounts
+
+A standard approach to authenticate users with Algorand accounts.
+
+## Abstract
+
+This ARC introduces a standard for SSO authentication based on Algorand accounts. It leverages the public-secret key <*PK, SK*> encryption schema to verify the identity of a user that owns an Algorand account. This approach fosters the adoption of novel identity management systems for Web3 applications.
```

Comment:
There is not explicit here that covers SSO, which is about server->server trust relationships. I am not clear how I as a service provider could trust an identity provider using the described mechanisms

Reply:
I agree that SSO could be misinterpreted here. What I meant here is the following: Imagine an application composed by an on-chain and off-chain component.
- The on chain part requires users to connect with their account to interact, e.g. with a smart contract;
- The off-chain part requires a user to login with credentials (e.g. user-pwd).

With this ARC the application can authenticate the user only once for both components. SSO across on-chain/off-chain part.

Reply:
OK but that's so far from what SSO normally means, it's worth removing. SSO is about server to server trust configurations such that a single 'identity provider' can issue tokens and dependent 'relying parties' be able to accept those tokens (eg, via signature using a trusted server certificate).

Regarding the scenario above with offline verifiers then it seems to me that can only work if rekeying the account is somehow prevented during the time the verifier is offline, I think. It seems to me that adding some mechanism to help with that is doable.